src/backend/vulkan/src/window.rs

```diff
@@ -16,13 +16,102 @@ use winit;
 use conv;
 use {VK_ENTRY, Backend, Instance, PhysicalDevice, QueueFamily, RawInstance};
 
+pub trait GetExtentable {
+    fn get_extent(&self) -> Option<(u32, u32)>;
+}
+
+#[cfg(feature = "winit")]
+impl GetExtentable for winit::Window {
+    fn get_extent(&self) -> Option<(u32, u32)> {
+        self.get_inner_size()
+    }
+}
+
+enum Connection {
+    #[cfg(all(unix, not(target_os = "android")))]
+    Xlib(*mut vk::Display, vk::Window),
+    #[cfg(all(unix, not(target_os = "android")))]
+    Xcb(*mut vk::xcb_connection_t, vk::xcb_window_t),
+    #[cfg(all(unix, not(target_os = "android")))]
+    Wayland(Arc<GetExtentable>),
```

Comment:
why are these trait objects instead of concrete types like the other variants?

Reply:
So we can support non-winit.